### PR TITLE
AWS: Add Service endpoints to VPC

### PR DESCRIPTION
**Pull request by ameukam**
Add service endpoints for AWS Services used in kops-ci account.

### CODE REVIEW

**Comment by ameukam** [comment]
/hold
cc @dims @bryantbiggs

**Review by bryantbiggs — APPROVED** [review]
LGTM!

File: infra/aws/terraform/kops-infra-ci/vpc.tf [hunk 1]
```diff
@@ -101,3 +101,51 @@ module "vpc" {
...
+    s3 = {
+      service         = "s3"
+      service_type    = "Gateway"
+      route_table_ids = module.vpc.private_route_table_ids
+      tags = {
+        Name = "${local.prefix}-s3"
+      }
+    }
+    },
+    { for service in toset(["autoscaling", "ecr.api", "ecr.dkr", "ec2", "ec2messages", "elasticloadbalancing", "sts", "kms", "logs", "ssm", "ssmmessages"]) :
```
Comment (micahhausler) [inline infra/aws/terraform/kops-infra-ci/vpc.tf:140]: Are pods in the VPC going to use pod identity? You'll need to add `eks-auth` if so

Comment (ameukam) [reply infra/aws/terraform/kops-infra-ci/vpc.tf]: @micahhausler updated.

File: infra/aws/terraform/kops-infra-ci/vpc.tf [hunk 2]
```diff
@@ -101,3 +101,51 @@ module "vpc" {
...
+    s3 = {
+      service         = "s3"
+      service_type    = "Gateway"
+      route_table_ids = module.vpc.private_route_table_ids
+      tags = {
+        Name = "${local.prefix}-s3"
+      }
+    }
+    },
+    { for service in toset(["autoscaling", "ecr.api", "ecr.dkr", "ec2", "ec2messages", "eks-auth", "elasticloadbalancing", "sts", "kms", "logs", "ssm", "ssmmessages"]) :
```
Comment (upodroid) [inline infra/aws/terraform/kops-infra-ci/vpc.tf:140]:
```suggestion
    { for service in toset(["autoscaling", "ecr.api", "ecr.dkr", "ec2", "ec2messages", "eks-auth", "elasticloadbalancing", "events", "sts", "sqs", "kms", "logs", "ssm", "ssmmessages"]) :
```

kops creates EventBridge and SQS resources

https://storage.googleapis.com/kubernetes-jenkins/logs/e2e-ci-kubernetes-e2e-al2023-aws-conformance-canary/1731732032918130688/build-log.txt

Comment (ameukam) [reply infra/aws/terraform/kops-infra-ci/vpc.tf]: Added

**Comment by ameukam** [comment]
VPC Endpoints created:

```bash
aws ec2 describe-vpc-endpoints --region us-east-2 --query 'VpcEndpoints[*].ServiceName' --output table
--------------------------------------------------
|              DescribeVpcEndpoints              |
+------------------------------------------------+
|  com.amazonaws.us-east-2.s3                    |
|  com.amazonaws.us-east-2.ecr.dkr               |
|  com.amazonaws.us-east-2.ec2                   |
|  com.amazonaws.us-east-2.sqs                   |
|  com.amazonaws.us-east-2.sts                   |
|  com.amazonaws.us-east-2.ssm                   |
|  com.amazonaws.us-east-2.eks-auth              |
|  com.amazonaws.us-east-2.kms                   |
|  com.amazonaws.us-east-2.elasticloadbalancing  |
|  com.amazonaws.us-east-2.ec2messages           |
|  com.amazonaws.us-east-2.events                |
|  com.amazonaws.us-east-2.ecr.api               |
|  com.amazonaws.us-east-2.ssmmessages           |
|  com.amazonaws.us-east-2.autoscaling           |
|  com.amazonaws.us-east-2.logs                  |
+------------------------------------------------+
```

**Comment by ameukam** [comment]
Ref:
  - https://github.com/kubernetes/k8s.io/issues/5127

**Pull request merged**